pkg/mesh/topology.go

```diff
@@ -173,9 +177,37 @@ func NewTopology(nodes map[string]*Node, peers map[string]*Peer, granularity Gra
 		}
 	}
 
+	// Copy the host node DiscoveredEndpoints first.
+	for key := range nodes[hostname].DiscoveredEndpoints {
+		t.discoveredEndpoints[key] = nodes[hostname].DiscoveredEndpoints[key]
+	}
+	// Now that the topology is ordered, update the discoveredEndpoints map
+	// add new ones by going through the ordered topology: segments, nodes
+	for _, segment := range t.segments {
+		for _, node := range topoMap[segment.location] {
```

Comment:
small optimization: can we move this inside of this loop https://github.com/squat/kilo/blob/d1948acd770732ef71353644d137cddebaac26c4/pkg/mesh/topology.go#L164-L174

Reply:
done